src/module/dialog-flow.module.ts

```diff
@@ -18,10 +29,52 @@ export class DialogFlowModule implements NestModule {
 
 		return {
 			module: DialogFlowModule,
+			providers: [DialogFlowService],
 			controllers: [DialogFlowController.forRoot(webHookConfig)],
 		};
 	}
 
+	constructor(
+		private readonly modulesContainer: ModulesContainer,
+		private readonly moduleRef: ModuleRef, 
+		private readonly dialogFlowService: DialogFlowService,
+	) {}
+
+	public onModuleInit(): any {
+		const metadataScanner = new MetadataScanner();
+		const modules = [...this.modulesContainer.values()];
+
+		modules.forEach(({ metatype }) => {
+			const metadata = Reflect.getMetadata(nestMetadata.COMPONENTS, metatype) || [];
+			const components = [...metadata.filter(metatype => typeof metatype === 'function')];
+
+			components.map(component => {
+				const reflectedMetadata = metadataScanner
+					.scanFromPrototype(null, component.prototype, method => {
+						const intentOrAction =
+							Reflect.getMetadata(
+								DIALOG_FLOW_INTENT,
+								Reflect.getOwnPropertyDescriptor(component.prototype, method).value,
+							) ||
+							Reflect.getMetadata(
+								DIALOG_FLOW_ACTION,
+								Reflect.getOwnPropertyDescriptor(component.prototype, method).value,
+							);
+						return intentOrAction ? { provider: component, intentOrAction, method } : null;
+					})
+					.filter(v => v);
+
+				[...reflectedMetadata].forEach(metadata => {
+					this.dialogFlowService.addHandler(
```

Comment:
for the handlers management, if you could create something like this [https://github.com/adrien2p/nestjs-mandrill/blob/master/src/core/handlers.container.ts](https://github.com/adrien2p/nestjs-mandrill/blob/master/src/core/handlers.container.ts) instead of keeping it into the service and separate the responsability

Reply:
Then you will be able to call the container to check if something already exist for the intent/action and throw in that case.